draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+
+Defining a default base IRI as allowed by {{!RFC3986, Section 5.1.4}} is deferred to context specifications.
+
+### Identifiers vs Locators {#id-vs-loc}
+
+IRIs used in JRI keywords are identifiers (IRIs/URIs) and not necessarily locators (IRLs/URLs), even if they specify a scheme indicating a protocol suitable for resource retrieval.  Context specifications SHOULD indicate under what circumstances a reference can be treated as a locator and retrieved on demand, and address the security concerns related to automatic retrieval.
+
+Standalone implementations of context-independent JRI MUST NOT attempt to automatically retrieve references by default.  They MAY offer a configuration option to do so, and MAY offer a mechanism to register handlers based on the IRI scheme or other IRI components.
+
+### Duplicate IRIs {#duplicate-iris}
+
+A resource MAY (and likely will) have multiple IRIs, but there is no way for an IRI to identify more than one resource.  When multiple primary or secondary resources attempt to identify as the same IRI through any combination of JRI keywords and context specification features, implementations SHOULD raise an error condition.  Otherwise the result is undefined, and even if documented will not be interoperable.
+
+## JRI Keywords
+
+JRI defines keywords in three behavior categories:
+
+* Identifiers, which assign IRIs to primary or secondary resources and (for primary resources) set the resource's base IRI
+* Locations, which indicate which locations are referenceable and can contain identifiers
+* References, which link one part of a document to another part of the same document or to a different document
+
+Context specifications MAY define their own keywords or other mechanisms within these three categories, as discussed in {{context}}.  They also MAY restrict reference targets to those identified by JRI and/or context specification identifier or location keywords.
+
+### General syntax
+
+JRI keyword names and string values are defined to be representable as {{?RFC3629}} UTF-8 strings, with keyword names representable in {{?RFC20}} ASCII.  As such, JRI keywords can be used as object properties in interoperable JSON documents as prescribed by {{?RFC8259, Sections 4, 7, and 8.1}}, as well as in any other JSON Pointer-compatible data formats.
+
+JRI keywords MAY be used in data formats that are only capable of representing a subset of UTF-8 string values, with the obvious limitation that only the representable subset of JRI keyword values will be usable.
+
+As {{!RFC3987, Section 3}} defines mappings between IRI and URI syntax, JRI MAY be used in data formats that can only directly represent URIs.
+
+#### The "$" prefix
+
+JRI keywords MUST start with a "$" character.  Context specifications MAY define their own "$"-prefixed keywords, however this is NOT RECOMMENDED as future specifications that extend JRI as discussed in {{extending}} will use that prefix.[^8]
```

Comment:
@DavidBiesack Can you open an issue for this?  And if possible give some use cases for it?  I'm not against this, but I think it warrants a more visible discussion of its own than it will get in this PR.

Reply:
Thanks - see https://github.com/json-schema-org/referencing/issues/6